signer/src/config/serialization.rs

```diff
@@ -33,6 +35,39 @@ where
         .map_err(serde::de::Error::custom)
 }
 
+/// A deserializer for the std::time::Duration type.
+/// Serde includes a default deserializer, but it expects a struct.
+pub fn duration_seconds_deserializer<'de, D>(
+    deserializer: D,
+) -> Result<std::time::Duration, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    Ok(std::time::Duration::from_secs(
+        u64::deserialize(deserializer).map_err(serde::de::Error::custom)?,
+    ))
+}
+
+/// A deserializer for Bitcoin processing delay, capping its max value.
+pub fn bitcoin_processing_delay_deserializer<'de, D>(
+    deserializer: D,
+) -> Result<std::time::Duration, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let delay = duration_seconds_deserializer(deserializer)?;
+    let delay_secs = delay.as_secs();
+    match delay_secs > MAX_BITCOIN_PROCESSING_DELAY_S {
+        true => Err(serde::de::Error::custom(
```

Comment:
So we're generally not considering "valid" invalid values as errors during deserialization, so I think you could get rid of this custom serializer and just use the `duration_seconds_deserializer`.

There's `impl Validatable for SignerConfig {` on line 245 where the validation should be plugged in. I think that might make this all a little cleaner :)